virtcontainers/api.go

```diff
@@ -45,23 +45,30 @@ func createSandboxFromConfig(sandboxConfig SandboxConfig) (*Sandbox, error) {
 		return nil, err
 	}
 
+	defer func() {
+		if err != nil {
+			s.Logger().WithError(err).WithField("sandboxid", s.id).Error("Creating sandbox failed")
+			s.Cleanup()
```

Comment:
I am not sure if we can simply apply one big cleanup function because the failure could be caused by a bunch of different reasons, and using a hammer to cleanup might end up with some conflicts.
Trying to be more concrete here, let's say this function fails when creating the network, we don't want to call into `s.stop()` because this will try to communicate with the agent to stop the sandbox inside the VM, but the VM does not even exist at this point. That's why I think something with finer granularity would be more appropriate.